cmake/sysbuild/suit.cmake

```diff
@@ -528,19 +527,11 @@ function(suit_setup_merge)
     list(APPEND ARTIFACTS_TO_MERGE ${IMAGE_BINARY_DIR}/zephyr/${IMAGE_BINARY_FILE}.hex)
     list(APPEND ARTIFACTS_TO_MERGE ${IMAGE_BINARY_DIR}/zephyr/uicr.hex)
 
-    unset(CONFIG_SUIT_ENVELOPE_OUTPUT_MPI_MERGE)
-    sysbuild_get(CONFIG_SUIT_ENVELOPE_OUTPUT_MPI_MERGE IMAGE ${image} VAR CONFIG_SUIT_ENVELOPE_OUTPUT_MPI_MERGE KCONFIG)
-    if(CONFIG_SUIT_ENVELOPE_OUTPUT_MPI_MERGE)
-      list(APPEND ARTIFACTS_TO_MERGE ${BINARY_DIR}/zephyr/suit_mpi_${IMAGE_TARGET_NAME}_merged.hex)
-    endif()
-
-    if(${IMAGE_TARGET_NAME} STREQUAL "application")
-      # Get a list of files (and their dependencies) which need merging into the uicr merged file
-      # and add them at the end of the list, allowing for overwriting, this only applies to the
-      # application core image
-      get_property(merge_files GLOBAL PROPERTY SUIT_MERGE_FILE)
-      get_property(merge_dependencies GLOBAL PROPERTY SUIT_MERGE_DEPENDENCIES)
-    else()
+    # Get a list of files (and their dependencies) which need merging into the uicr merged file
+    # and add them at the end of the list, allowing for overwriting
+    get_property(merge_files GLOBAL PROPERTY SUIT_MERGE_${IMAGE_TARGET_NAME}_FILE)
```

Comment:
I think `unset(merge_files)` is needed for the if clause below to work properly - or at least it is a bit cleaner